### PR TITLE
Refactored return if pattern

### PR DESCRIPTION
**Pull request by Vitaly-Protasov**

### CODE REVIEW

File: test/patterns/if_return_if_detection/test_if_return_if.py [hunk 1]
```diff
@@ -30,14 +30,14 @@ class TestCountIfReturn(TestCase):
...
 
-    def test_no_return_inside(self):
-        lines = self.countifreturn.value(Path(self.dir_path, '2.java'))
-        self.assertEqual(lines, [])
-
     def test_2nd_level_inside(self):
         lines = self.countifreturn.value(Path(self.dir_path, '1.java'))
         self.assertEqual(lines, [6, 10])
 
+    def test_no_return_inside(self):
```
Comment (aravij) [inline test/patterns/if_return_if_detection/test_if_return_if.py:37]: +1

File: aibolit/patterns/if_return_if_detection/if_detection.py [hunk 2]
```diff
@@ -24,9 +11,15 @@ class CountIfReturn:
...
-        ''''''
-        tree = newJavalangImproved(filename)
-        if_decls = tree.filter([javalang.tree.IfStatement])
-
-        return if_decls
+    def value(self, filename: str) -> List[int]:
+        detected_lines = []
+        p = JavaPackage(filename)
+        for class_name in p.java_classes:
+            tree = p.java_classes[class_name]
```
Comment (aravij) [inline aibolit/patterns/if_return_if_detection/if_detection.py:18]: We better call it as is - `java_class`.

File: aibolit/patterns/if_return_if_detection/if_detection.py [hunk 3]
```diff
@@ -24,9 +11,15 @@ class CountIfReturn:
...
 
-    def value(self, filename: str):
-        ''''''
-        tree = newJavalangImproved(filename)
-        if_decls = tree.filter([javalang.tree.IfStatement])
-
-        return if_decls
+    def value(self, filename: str) -> List[int]:
+        detected_lines = []
+        p = JavaPackage(filename)
```
Comment (aravij) [inline aibolit/patterns/if_return_if_detection/if_detection.py:16]: We shouldn't use single letter variables with large scope or with some complex objects. We better use `java_package` here.

File: aibolit/patterns/if_return_if_detection/if_detection.py [hunk 4]
```diff
@@ -24,9 +11,15 @@ class CountIfReturn:
...
+        for class_name in p.java_classes:
+            tree = p.java_classes[class_name]
+            for index, each_object in enumerate(tree.nodes_by_type(ASTNodeType.IF_STATEMENT)):
```
Comment (aravij) [inline aibolit/patterns/if_return_if_detection/if_detection.py:19]: `each_object` is meaningless. It is better to call it `if_node`.

File: aibolit/patterns/if_return_if_detection/if_detection.py [hunk 5]
```diff
@@ -24,9 +11,15 @@ class CountIfReturn:
...
+                all_childs = list([i for i in tree.tree.succ[each_object]])
```
Comment (aravij) [inline aibolit/patterns/if_return_if_detection/if_detection.py:20]: You don't need the outer `list`. list comprehension do this job.

File: test/patterns/if_return_if_detection/test_if_return_if.py (hunk 1 above)
Comment (Vitaly-Protasov) [reply test/patterns/if_return_if_detection/test_if_return_if.py:37]: nope, you are not right. To understant this idea in a proper way, you can read it: https://www.yegor256.com/2015/01/21/if-then-throw-else.html

File: aibolit/patterns/if_return_if_detection/if_detection.py (hunk 2 above)
Comment (Vitaly-Protasov) [reply aibolit/patterns/if_return_if_detection/if_detection.py:18]: Why do you think that it is better? To use java_class we need to initialize ast with parameters, such as tree, root, java_package. Hence, the work is only growth.

File: aibolit/patterns/if_return_if_detection/if_detection.py (hunk 3 above)
Comment (Vitaly-Protasov) [reply aibolit/patterns/if_return_if_detection/if_detection.py:16]: got it

File: aibolit/patterns/if_return_if_detection/if_detection.py (hunk 4 above)
Comment (Vitaly-Protasov) [reply aibolit/patterns/if_return_if_detection/if_detection.py:19]: agree

File: aibolit/patterns/if_return_if_detection/if_detection.py (hunk 5 above)
Comment (Vitaly-Protasov) [reply aibolit/patterns/if_return_if_detection/if_detection.py:20]: got it

**Comment by aravij** [comment]
@acheshkov  please merge

**Comment by acheshkov** [comment]
@rultor merge

**Comment by acheshkov** [comment]
@rultor merge

**Pull request merged**